### PR TITLE
Refactor and fix config changes bug

### PR DESCRIPTION
**Pull request by en0ma**
This PR is a refactor and improvement on the changes introduced by https://github.com/application-research/estuary/pull/141. 

It makes sure the changes introduced by 141 are a reflection of what was there prior to the changes.

I tested;
- making deals
- pinning and ipfs gateway retrieval

### CODE REVIEW

File: cmd/estuary-shuttle/main.go [hunk 1]
```diff
@@ -223,22 +227,27 @@ func main() {
 		},
 		&cli.BoolFlag{
 			Name:  "logging",
+			Usage: "enable api endpoint logging",
 			Value: cfg.LoggingConfig.ApiEndpointLogging,
 		},
 		&cli.BoolFlag{
 			Name:  "write-log-flush",
+			Usage: "enable hard flushing blockstore",
```
Comment (gmelodie) [inline cmd/estuary-shuttle/main.go:235]: Maybe we could make this wording better? I'm not sure what it means just by looking at it

File: cmd/estuary-shuttle/main.go [hunk 2]
```diff
@@ -223,22 +227,27 @@ func main() {
...
 			Value: cfg.NodeConfig.HardFlushWriteLog,
 		},
 		&cli.BoolFlag{
 			Name:  "write-log-truncate",
+			Usage: "enables write log truncating",
```
Comment (gmelodie) [inline cmd/estuary-shuttle/main.go:240]: nit:
```suggestion
			Usage: "truncates old logs with new ones",
```

File: cmd/estuary-shuttle/main.go [hunk 3]
```diff
@@ -223,22 +227,27 @@ func main() {
...
 			Value: cfg.NodeConfig.WriteLogTruncate,
 		},
 		&cli.BoolFlag{
 			Name:  "no-blockstore-cache",
+			Usage: "disable blockstore caching",
 			Value: cfg.NodeConfig.NoBlockstoreCache,
 		},
 		&cli.BoolFlag{
 			Name:  "private",
+			Usage: "sets shuttle as private",
```
Comment (gmelodie) [inline cmd/estuary-shuttle/main.go:250]: More of a question: what does it mean for a shuttle to be private? Maybe we could make the wording of this explain what it is? If it's kind of a complex explanation nvm

File: cmd/estuary-shuttle/main.go [hunk 4]
```diff
@@ -248,6 +257,7 @@ func main() {
 		},
 		&cli.BoolFlag{
 			Name:  "no-reload-pin-queue",
+			Usage: "disable reloading pins on shuttle start",
```
Comment (gmelodie) [inline cmd/estuary-shuttle/main.go:260]: nit:
```suggestion
			Usage: "disable reloading pin queue on shuttle start",
```

File: cmd/estuary-shuttle/main.go [hunk 5]
```diff
@@ -275,7 +287,8 @@ func main() {
 		},
 		&cli.BoolFlag{
 			Name:  "libp2p-websockets",
-			Value: cfg.NodeConfig.HasListener(config.DefaultWebsocketAddr),
+			Usage: "enable adding libp2p websockets listen addr",
```
Comment (gmelodie) [inline cmd/estuary-shuttle/main.go:290]: Another question: what does this mean? Can we make the wording more intuitive?

File: config/estuary.go [hunk 6]
```diff
@@ -34,27 +34,32 @@ func (cfg *Estuary) Save(filename string) error {
...
+	}
+	return nil
+}
 
-	cfg := Estuary{
-		DataDir:                pwd,
-		StagingDataDir:         filepath.Join(pwd, "stagingdata"),
+func NewEstuary() *Estuary {
+	return &Estuary{
+		DataDir:                ".",
```
Comment (gmelodie) [inline config/estuary.go:54]: I'm not sure that's what we want as default. If we're running estuary and our `pwd` is `/home/gabe` but my estuary folder and bin are at `/home/gabe/estuary`, I would want the default folder to be `/home/gabe/estuary`.

File: main.go [hunk 7]
```diff
@@ -156,23 +156,33 @@ func (s *Server) updateAutoretrieveIndex(tickInterval time.Duration, quit chan s
...
+		case "no-blockstore-cache":
+			cfg.NodeConfig.NoBlockstoreCache = cctx.Bool("no-blockstore-cache")
 		case "write-log-truncate":
 			cfg.NodeConfig.WriteLogTruncate = cctx.Bool("write-log-truncate")
+		case "write-log-flush":
+			cfg.NodeConfig.HardFlushWriteLog = cctx.Bool("write-log-flush")
 		case "write-log":
-			cfg.NodeConfig.WriteLogDir, err = config.MakeAbsolute(cfg.DataDir, cctx.String("write-log"))
+			if wl := cctx.String("write-log"); wl != "" {
+				if wl[0] == '/' {
```
Comment (gmelodie) [inline main.go:180]: Again, maybe use https://pkg.go.dev/strings#HasPrefix here

File: main.go [hunk 8]
```diff
@@ -333,14 +353,26 @@ func main() {
...
+			Value: cfg.NodeConfig.Blockstore,
 		},
 		&cli.BoolFlag{
 			Name:  "write-log-truncate",
+			Usage: "enables log truncating",
 			Value: cfg.NodeConfig.WriteLogTruncate,
 		},
+		&cli.BoolFlag{
+			Name:  "write-log-flush",
+			Usage: "enable hard flushing blockstore",
```
Comment (gmelodie) [inline main.go:365]: Same comment about hard flushing above

File: main.go [hunk 9]
```diff
@@ -333,14 +353,26 @@ func main() {
...
+			Value: cfg.NodeConfig.HardFlushWriteLog,
+		},
+		&cli.BoolFlag{
+			Name:  "no-blockstore-cache",
+			Usage: "disable blockstore caching",
```
Comment (gmelodie) [inline main.go:370]: Same comment about caching above

**Review by gmelodie — APPROVED** [review]
LGTM. I have a few nitpicks, but feel free to ignore them if you don't agree :)

File: cmd/estuary-shuttle/main.go [hunk 10]
```diff
@@ -77,34 +77,37 @@ func init() {
...
 			continue
 		}
+
 		switch name {
 		case "datadir":
-			cfg.SetDataDir(cctx.String("datadir"))
+			cfg.DataDir = cctx.String("datadir")
 		case "blockstore":
-			cfg.NodeConfig.BlockstoreDir = cctx.String("blockstore")
+			cfg.NodeConfig.Blockstore = cctx.String("blockstore")
```
[the remark this answers is not in the document]
Comment (en0ma) [reply cmd/estuary-shuttle/main.go:92]: The thing is, it is not a dir, it is more of config string

File: config/estuary.go (hunk 6 above)
Comment (en0ma) [reply config/estuary.go:54]: This PR is to make changes to how things were, just before `pwd` was introduced. Also, `"."` works as expected when `ddir` is not specified

File: cmd/estuary-shuttle/main.go (hunk 5 above)
Comment (en0ma) [reply cmd/estuary-shuttle/main.go:290]: This checks if the libp2p websockets addr should be added to the listen addrs set.

File: main.go (hunk 9 above)
Comment (en0ma) [reply main.go:370]: if true, the blockstore will not be wrapped in a caching layer

File: cmd/estuary-shuttle/main.go (hunk 3 above)
Comment (en0ma) [reply cmd/estuary-shuttle/main.go:250]: I honestly can not give another explanation besides making a shuttle private - other cannot connect to it or something. I added these simple comments as non where there in the first place

**Pull request merged**